js/src/components/tree-select-control/control.js

```diff
@@ -31,6 +32,7 @@ const Control = ( {
 	disabled,
 	onFocus = () => {},
 	onTagsChange = () => {},
+	onSearch = () => {},
```

Comment:
Got it.

In terms of the internal relationship between these components, I was thinking the `onSearch` of  \<Control\> seems to be much close to a `onChange` callback from the perspective of \<TreeSelectControl\>. Because it only forwards the callback to the `onChange` of \<input\> without doing anything related to search.

In addition, the relevant JSDoc describes the *change* behavior rather than *search*:
- https://github.com/woocommerce/google-listings-and-ads/blob/f222e095f1c3aa94d496283990b22f3621a31767/js/src/components/tree-select-control/control.js#L24
- https://github.com/woocommerce/google-listings-and-ads/blob/f222e095f1c3aa94d496283990b22f3621a31767/js/src/components/tree-select-control/index.js#L299

So, perhaps it could be considered using the `onChange` instead.